pkg/pillar/cmd/zedrouter/appnumonunet.go

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Network Instance/underlay network IP Address Management,
+// App Number management Module
+
+package zedrouter
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+// Network Instance Application number/IP Address Management
+// ---------------------------------------------------------
+//  AppNum  Module (bitMap & persistent AppNumMap)
+//   - uNetAppNumMapGet()
+//   - uNetAppNumGet()
+//   - uNetAppNumAlloc()
+//   - uNetAppNumDelete()
+//   - networkInstanceAppNumMapInitialize()
+//   - networkInstanceAppNumMapDelete()
+//   - getDhcpAppNumCountOnDhcpRange()
+//   - getDhcpAppNumCountOnSubnet()
+//   - networkInstanceDhcpAppNumCountMax()
+//  BitMap : is for the Dynamic dhcp address/app number assignments
+// For BitMap:(DhcpRange.Start - DhcpRange.End)
+//    - bitMapAppNumAdd(), set a bit in the BitMap
+//    - bitMapAppNumDelete(), reset a bit in the BitMap
+
+// mapped on NetworkInstance ID
+var netInstAppNumMap map[string]*types.NetworkInstanceAppNumMapInfo
```

Comment:
Why do we need this optimization? Simpler (hence less bugs) to just do a Get on the persistent publication. We don't need a map for the network instance here.